platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/alignment/AlignmentIntegrationTest.groovy

```diff
@@ -451,7 +451,7 @@ class AlignmentIntegrationTest extends AbstractAlignmentSpec {
         expectAlignment {
             module('core') alignsTo('2.9.4') byPublishedPlatform()
             module('databind') tries('2.7.9') alignsTo('2.9.4') byPublishedPlatform()
-            module('annotations') tries('2.7.9') alignsTo('2.9.4') byPublishedPlatform()
+            module('annotations') tries('2.7.9') tries('2.9.0') alignsTo('2.9.4') byPublishedPlatform()
```

Comment:
These changes are caused by no longer eagerly removing outgoing edges from losing nodes of version/module/capability conflicts. We still remove the outgoing edges of these nodes when they are processed when they're removed from the queue (we onFewerSelected them when we retarget edges away from them), but new ordering causes us to try a few different versions here.

The advantage of this new approach is that we no longer are susceptible to reentrant module selection, preventing concurrent modification exceptions during module selection.